bambi/plots/effects.py

```diff
@@ -0,0 +1,402 @@
+        Transformations that are applied to each of the variables being plotted. The keys are the
+        name of the variables, and the values are functions to be applied. Defaults to ``None``.
+
+    Returns
+    -------
+    pandas.DataFrame
+        A dataframe with the comparison values, highest density interval, contrast name,
+        contrast value, and conditional values.
+
+    Raises
+    ------
+    ValueError
+        If length of ``contrast`` is greater than 1.
+        If ``contrast`` is not a string, dictionary, or list.
+        If ``comparison_type`` is not 'diff' or 'ratio'.
+        If ``prob`` is not > 0 and < 1.
+    """
+
+    if not isinstance(contrast, (dict, list, str)):
+        raise ValueError("'contrast' must be a string, dictionary, or list.")
+    if isinstance(contrast, (dict, list)):
+        if len(contrast) > 1:
+            raise ValueError(
+                f"Only one contrast predictor can be passed. {len(contrast)} were passed."
+            )
+
+    if comparison_type not in ("diff", "ratio"):
+        raise ValueError("'comparison_type' must be 'diff' or 'ratio'")
+
+    if prob is None:
+        prob = az.rcParams["stats.hdi_prob"]
+    if not 0 < prob < 1:
+        raise ValueError(f"'prob' must be greater than 0 and smaller than 1. It is {prob}.")
+
+    comparison_functions = {"diff": lambda x, y: x - y, "ratio": lambda x, y: x / y}
+    lower_bound = round((1 - prob) / 2, 4)
+    upper_bound = 1 - lower_bound
+
+    contrast_info = ContrastInfo(model, contrast)
+    conditional_info = ConditionalInfo(model, conditional)
+
+    # 'comparisons' should not be restricted to ("main", "group", "panel")
+    comparisons_df = create_comparisons_data(
+        conditional_info, contrast_info, user_passed=conditional_info.user_passed
+    )
+
+    if transforms is None:
+        transforms = {}
+
+    response_name = get_aliased_name(model.response_component.response_term)
+    response = Response(response_name)
+
+    # perform predictions on new data
+    idata = model.predict(idata, data=comparisons_df, inplace=False)
+
+    def _compute_contrast_estimate(
+        contrast: ContrastInfo,
+        response: Response,
+        comparisons_df: pd.DataFrame,
+        idata: az.InferenceData,
+    ) -> ContrastEstimate:
+        """
+        Computes the contrast comparison estimate and highest density interval
+        for a given contrast and response by first subsetting posterior draws
+        using a contrast mask. Then, pairwise comparisons are computed for the
+        contrast values. Finally, the mean comparison and lower/upper bounds
+        are computed for each pairwise comparison.
+        """
+        function = comparison_functions[comparison_type]
+
+        draws = {}
+        for idx, val in enumerate(contrast.values):
+            mask = np.array(comparisons_df[contrast.name] == contrast.values[idx])
+            select_draw = idata.posterior[f"{response.name}_{response.target}"].sel(
```

Comment:
This will have a similar conflict as the one mentioned above when using aliases for non-parent parameters.

Reply:
I replied to your comment above regarding "non-parent parameters"

Reply:
Now that I remember, we previously discussed that for this first PR, `plot_comparisons` will not have the functionality to plot and compare non-parent parameters. Therefore there is no `target` arg. and no possibility of a `KeyError` being thrown since we use `get_aliased_name` for the response.